include/GroverSimulator.hpp

```diff
@@ -53,19 +54,19 @@ class GroverSimulator: public Simulator<dd::Package<>> {
         }
     }
 
-    [[nodiscard]] dd::QubitCount getNumberOfQubits() const override { return n_qubits + 1; };
+    [[nodiscard]] std::size_t getNumberOfQubits() const override { return n_qubits + 1; };
 
     [[nodiscard]] std::size_t getNumberOfOps() const override { return 0; };
 
     [[nodiscard]] std::string getName() const override { return "emulated_grover_" + std::to_string(n_qubits); };
 
-    [[nodiscard]] std::string getOracle() const { return oracle; }
+    [[nodiscard]] std::string getOracle() const { return {oracle.rbegin(), oracle.rend()}; }
```

Comment:
Because for strings the character at position 0 is the left-most one, whereas with our convention on qubit order, it should be the right-most (least significant one).

Reply:
Makes sense 👍🏻
Funny how such oversights can slip through over the years.
Feel free to resolve this conversation.